src/analysis/multiSpecies/Microbiome_Modeling_Toolbox/PairwiseInteractionModeling/simulatePairwiseInteractions.m

```diff
@@ -0,0 +1,375 @@
+% saveSolutionsFlag    If true, flux solutions are stored (may result in 
+%                      large output files)
+% numWorkers           Number of workers in parallel pool if desired
+% sigD                 Difference in growth rate that counts as significant
+%                      change (by default: 10%)
+% OUTPUTS
+% pairwiseInteractions     Table with computed pairwise and single growth
+%                          rates for all entered microbe-microbe models
+% OPTIONAL OUTPUT
+% pairwiseSolutions        Table with all computed flux solutions saved
+%                         (may result in large output files)
+%
+% .. Authors:
+%       - Almut Heinken, 02/2018 
+
+% if no diet was entered
+if nargin <2
+    inputDiet=[];
+end
+
+if nargin >3 && saveSolutionsFlag==true
+    pairwiseSolutions{1,1}='pairedModelID';
+    pairwiseSolutions{1,2}='PairwiseSolution';
+    pairwiseSolutions{1,3}='SingleModel1Solution';
+    pairwiseSolutions{1,4}='SingleModel2Solution';
+end
+
+if nargin>4 && numWorkers>0
+    poolobj=gcp('nocreate');
+    if isempty(poolobj)
+        parpool(numWorkers)
+    end
+end
+
+% if sigD was not entered
+if nargin<6
+    sigD=0.1;
+end
+
+pairwiseSolutions={};
+pairwiseInteractions{1, 1} = 'pairedModelID';
+pairwiseInteractions{1, 2} = 'ModelID1';
+pairwiseInteractions{1, 3} = 'ModelID2';
+pairwiseInteractions{1, 4} = 'pairedGrowth_Model1';
+pairwiseInteractions{1, 5} = 'pairedGrowth_Model2';
+pairwiseInteractions{1, 6} = 'singleGrowth_Model1';
+pairwiseInteractions{1, 7} = 'singleGrowth_Model2';
+pairwiseInteractions{1, 8} = 'Outcome_Model1';
+pairwiseInteractions{1, 9} = 'Outcome_Model2';
+pairwiseInteractions{1, 10} = 'Total_Outcome';
+
+if nargin>4 && numWorkers>0
+    % with parallelization
+    solutionPairedTemp={};
+    solutionSingle1Temp={};
+    solutionSingle2Temp={};
+    iAFirstTemp={};
+    iASecondTemp={};
+    iABothTemp={};
+    global CBT_LP_SOLVER
+    solver = CBT_LP_SOLVER;
+    parfor i = 1:size(pairedModelInfo, 1)
+        changeCobraSolver(solver,'LP');
+        % load the model
+        pairedModel=inputModels{i};
+        % if a diet was input
+        if ~isempty(inputDiet)
+            pairedModel = useDiet(pairedModel, inputDiet);
+        end
+        % for each paired model, set both biomass objective functions as
+        % objectives
+        biomass1 = strcat(pairedModelInfo{i, 2}, '_', pairedModelInfo{i, 3});
+        biomass2 = strcat(pairedModelInfo{i, 4}, '_', pairedModelInfo{i, 5});
+        model1biomass = find(ismember(pairedModel.rxns, biomass1));
+        pairedModel.c(model1biomass, 1) = 1;
+        model2biomass = find(ismember(pairedModel.rxns, biomass2));
+        pairedModel.c(model2biomass, 1) = 1;
+        % enforce minimal growth
+        pairedModel = changeRxnBounds(pairedModel, biomass1, 0.001, 'l');
+        pairedModel = changeRxnBounds(pairedModel, biomass2, 0.001, 'l');
+        % calculate joint biomass
+        solutionPaired = solveCobraLP(pairedModel);
```

Comment:
This should be `optimizeCbModel`. `solveCobraLP` should be used only with actual LP problems, not with models.